precovery/ingest/index_observations.py

```diff
@@ -0,0 +1,151 @@
+import argparse
+import datetime
+import glob
+import logging
+import multiprocessing
+import os
+
+import newrelic.agent
+
+from precovery.config import DefaultConfig
+from precovery.precovery_db import PrecoveryDatabase
+
+logger = logging.getLogger()
+
+
+def index_individual_file(
+    db_dir,
+    filename,
+    dataset_id,
+    dataset_name,
+    reference_doi,
+    documentation_url,
+    sia_url,
+):
+    """
+    Loads a single calendar month dataset file into the database and filesystem.
+
+    Note:
+        This function only works because the files are pre-divided by dataset/calendar month.
+        This means that there should be no chance of collision writing to the binaries files on disk.
+    """
+    application = newrelic.agent.application()
+    with newrelic.agent.BackgroundTask(application, name="index_individual_file"):
+        logger = multiprocessing.get_logger()
+        logger.info(f"Indexing {filename}")
+
+        # Initialize the database client within the worker.
+        db = PrecoveryDatabase.from_dir(db_dir, mode="w")
+        db.frames.load_hdf5(
+            filename,
+            dataset_id,
+            0,
+            name=dataset_name,
+            reference_doi=reference_doi,
+            documentation_url=documentation_url,
+            sia_url=sia_url,
+        )
+        logger.info(f"Indexed {filename}")
+
+
+def index(
+    data_dir="data",
+    out_dir="database",
+    dataset_id=None,
+    nside=DefaultConfig.nside,
+    data_file_max_size=DefaultConfig.data_file_max_size,
+    dataset_name=None,
+    reference_doi=None,
+    documentation_url=None,
+    sia_url=None,
+):
+
+    logger = multiprocessing.get_logger()
```

Comment:
I didn't realize you could tap into multiprocessing's logger. This is cool!

Reply:
I think it's the only logger that works by default without a lot of headache.